server/src/resources/js/checkout.ts

```diff
@@ -0,0 +1,233 @@
+import successTemplate from './templates/successTemplate';
+import errorTemplate from './templates/errorTemplate';
+import $ from 'jquery';
+import { TYPES } from './types/types';
+import { inject, injectable } from 'inversify';
+import { GernzyGraphqlService } from './interfaces/graphqlService';
+import { GernzyCart } from './interfaces/cart';
+import { GernzyCheckout } from './interfaces/checkout';
+
+@injectable()
+class Checkout implements GernzyCheckout {
+    @inject(TYPES.GernzyGraphqlService) private graphqlService!: GernzyGraphqlService;
+    @inject(TYPES.GernzyCart) private cart!: GernzyCart;
+    private url!: string;
+
+    public endpointUrl(url: string) {
+        this.url = url;
+    }
+
+    public checkout() {
+        var userToken = localStorage.getItem('userToken') || '';
+        var self = this;
+        let query = `
+            query {
+                shopConfig {
+                    payment_providers
+                }
+            }
+        `;
+
+        window.checkoutForm = () => {
```

Comment:
Similarly let's experiment with de-nesting this too, just so it's easier to read.

Reply:
completed